packages/webview_flutter/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -79,6 +80,14 @@ public void onProgressChanged(WebView view, int progress) {
     }
   }
 
+  @VisibleForTesting
+  FlutterWebView(WebView webView, MethodChannel methodChannel, Handler platformThreadHandler) {
+    this.webView = webView;
+    this.methodChannel = methodChannel;
+    flutterWebViewClient = new FlutterWebViewClient(methodChannel);
+    this.platformThreadHandler = platformThreadHandler;
+  }
+
```

Comment:
Creating a separate constructor just for the purpose of testing is considered a bad practise (see also [this answer](https://softwareengineering.stackexchange.com/a/385805/336392)).

I have just merged my [PR](https://github.com/flutter/plugins/pull/4178) which refactors the existing constructor which should help you mock the `methodChannel` and `webView` instances more easily. If you merge with the master branch you should be able to remove this constructor and fix the unit-test.